src/scripts/main.js

```diff
@@ -1,5 +1,28 @@
 'use strict';
 
 document.addEventListener('click', e => {
-  // write code here
+  const wall = document.querySelector('.wall');
+  const spider = document.querySelector('.spider');
+
+  function setCoord(coord, outerSize, innerSize) {
+    const moveCoord = coord - innerSize / 2;
+
+    if (moveCoord < 0) {
+      return 0;
+    }
+
+    if (moveCoord > (outerSize - innerSize)) {
+      return outerSize - innerSize;
+    }
+
+    return moveCoord;
+  }
```

Comment:
move the function outside the event listener, now every time you click on the document the new function is created